sigstore/sign.py

```diff
@@ -30,7 +30,7 @@
 
 with artifact.open("rb") as a:
     signing_ctx = SigningContext.production()
-    with signing_ctx.with_signer(token, cache=True) as signer:
+    with signing_ctx.with_signer(token, fulcio=singing_ctx._fulcio, cache=True) as signer:
```

Comment:
We probably don't want to make the user pass the Fulcio instance back in like this 🙂 -- I left a suggestion below for making the `SigningContext` into an instance attribute on the `Signer`.

```suggestion
    with signing_ctx.with_signer(token, fulcio=singing_ctx._fulcio, cache=True) as signer:
```